iocage_lib/ioc_start.py

```diff
@@ -1312,19 +1311,27 @@ def __check_rtsold__(self):
             stdout=su.PIPE
         )
 
-    def get_default_gateway(self):
-        # e.g response - ('192.168.122.1', 'lagg0')
-        try:
-            return netifaces.gateways()["default"][netifaces.AF_INET]
-        except KeyError:
+    def get_default_interface(self):
+        if self.host_gateways['ipv4']['interface']:
+            return self.host_gateways['ipv4']['interface']
+        elif self.host_gateways['ipv6']['interface']:
+            return self.host_gateways['ipv6']['interface']
+        else:
             iocage_lib.ioc_common.logit(
                 {
                     'level': 'EXCEPTION',
-                    'message': 'No default gateway interface found'
+                    'message': 'No default interface found'
                 },
                 _callback=self.callback,
-                silent=self.silent
-            )
+                silent=self.silent)
+
+    def get_default_gateway(self, address_family='ipv4'):
+        if self.host_gateways[address_family]['gateway']:
+            return re.sub(
+                '%.*$', '%epair0b',
+                self.host_gateways[address_family]['gateway'])
+        else:
+            return None
```

Comment:
`None` as in the string `none` not python `None` ;) If you remember in an earlier PR you did we get the default gateway's value and if it's not `none` ( str one ), we set it else we don't. So this will fail there

Reply:
Oh yes: https://github.com/iocage/iocage/blob/43ba3fb55e557a5afcbb2a3bfb4288152ff04e49/iocage_lib/ioc_start.py#L955
I lost the big picture.